src/libostree/ostree-sepolicy.c

```diff
@@ -266,6 +270,56 @@ get_policy_checksum (char        **out_csum,
 
 #endif
 
+/**
+ * ostree_sepolicy_new_from_commit:
+ * @repo: The repo
+ * @commit: SHA-256 commit checksum
+ * @error: Error
+ *
+ * Extract the SELinux policy from a commit object via a partial checkout.  This is useful
+ * for labeling derived content as separate commits.
+ *
+ * This function is the backend of `ostree_repo_commit_modifier_set_sepolicy_from_commit()`.
+ *
+ * Returns: (transfer full): A new policy
+ */
+OstreeSePolicy*
+ostree_sepolicy_new_from_commit (OstreeRepo  *repo,
+                                 const char  *commit,
+                                 GCancellable *cancellable,
+                                 GError     **error)
+{
+  GLNX_AUTO_PREFIX_ERROR ("setting sepolicy from commit", error);
```

Comment:
`strjoina` hack to include commit hash?

Reply:
I've been trying to be more careful about adding checksums into error messages, because we now have several codepaths that log the checksum twice just inside ostree, and if the caller does it too the error message can become hard to read.